milli/src/update/settings.rs

```diff
@@ -1161,6 +1161,11 @@ impl InnerIndexSettingsDiff {
     pub fn settings_update_only(&self) -> bool {
         self.settings_update_only
     }
+
+    pub fn run_geo_indexing(&self) -> bool {
+        self.old.geo_fields_ids != self.new.geo_fields_ids
+            || (!self.settings_update_only && self.new.geo_fields_ids.is_some())
+    }
```

Comment:
Shouldn’t we ensure that we went from « not faceted » to « faceted » instead of checking if we're only updating the settings?
It seems more future-proof, but maybe I’m missing something? 🤔 

```suggestion
    pub fn run_geo_indexing(&self) -> bool {
        self.old.geo_fields_ids != self.new.geo_fields_ids
            || (!self.old.geo_fields_ids.is_none() && self.new.geo_fields_ids.is_some())
    }
```